test/functional/test-crypto-verifier.js

```diff
@@ -0,0 +1,126 @@
+describe('CryptoVerifier', function() {
+
+  const cryptoVerifier = installCryptoVerifierService(window);
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  const modulus =
+        '8iAq9Q7zaD93myumSS41aEha6HNRNlPD19tMQHHNjzFdtAH8-w' +
+        'T6poV3gsI7kfmg9vUrCSuaNrOqmF8HOfbNJfcHCKqXNMsgmYPH' +
+        'Wz3R7LStMCJ_476UnuMjA9LFSvrdO91_xaX4Jv0kpQ_mZw70wp' +
+        '_K6IFCcrAi85M__TKUwUpB5iXHy2M00vbSBw7ZbMUkMSELb-X1' +
+        '_sZbKczCtdYqJQfKqHRU4FxrGj3laRxafWDenpzeMoaSBOK7JK' +
+        'kILz8mr-eJ07bkKAfOUyOuTtCkPBzgPkaSvaEyqM-EA4VXP0gF' +
+        'IcI5HedIlWfX82yrxN-hZ0MbFrguuGL8iHX9w1aThHAmmyuX2V' +
+        '0A1Mh_c9dZtkZO0sYXcBOdh49qmkrn6n2hXP6mmcOF0caTOdke' +
+        'GQwR47zSut0MS4sJOyNmjkT4lqkLXb-1Ajva_PU-fTDRaIuxkr' +
+        'BPh0v2RwtwxNFytR9Ksrj_hwOYz_l81m8PnaBhMnkZorqF53bg' +
+        'eJ8jvDx2_mBb';
+  const pubExp = 'AQAB';
+
+  const rsaPubKey = {
+    'n': new Uint8Array(decodeStringToByteArray(modulus)),
+    'e': new Uint8Array(decodeStringToByteArray(pubExp))};
+  const modulus1 =
+        'vwx9LQrXHQmiVxxSK9IA_wq9Yu2TFDEQHk9b_rdYJe6fEEwron' +
+        '8W_9GBHbBwpNszMdgG531hZqvIU4xALhi47VsD4h5XJ2UNbCU2' +
+        'kpcJ-XC_Q62ArEY5vqxJgdYjq4bE8s3f8rKC-Uqg_uepoFEn-X' +
+        'Xf2l0UQmVcYCxeRY6ahvM';
+  const pubExp1 = 'AQAB';
+  const rsaPubKey1 = {
+    'n': new Uint8Array(decodeStringToByteArray(modulus1)),
+    'e': new Uint8Array(decodeStringToByteArray(pubExp1)),
+  };
+
+  const signature = new Uint8Array(hexToByteArray(
+      '0087de9f85b71386df784845d23103975591d73cd446212e184f1cb92ac6964f' +
+      'a2338d3832bb1c79e1b30eeaca6f24530c12286f26d73b969f6a8bc82ef24f6f' +
+      '70201fd2ff24d044b67dcfa067763fd9ef0437e707728f6e47160bf3b626cbbf' +
+      'dc6a61af1d802b7c57be496b7b768120c2135cb0a54fa1db5d1f74fe822aa400' +
+      'b04cee885bbaaffbbaea951bfeb2ac173698b1265755bc3ea8e1a389af2935e4' +
+      '6428d9114c0cff0c4531d76268765f40be76ccf470ac291b800ef8f08504d3ec' +
+      'da7675333e753d0a5de96c9a5fc052319134b32dc535a31f51382ddc24b946ac' +
+      '6134ea900af5a00ab7a528486f03273814b61207de1609864901f85f4f83ee74' +
+      '723802e9c2c7eacf3d1c2f8c6b5d4295b6ad3166a0c9f7482d494c9b8723add6' +
+      '2ce8d5868378c720a23cec6b80f802d1021323e3d22a57551de554d292d5fd64' +
+      '4de94fdb376b3e689e4d6239dac65bc4e9ae3265b10bfe87bb838aa15515b01e' +
+      '1cf21c14f99d474e12f5fcbeba8697074176d9bf8e3463d6195805c883710299' +
+      'e50f0eef74'));
+  const data = new Uint8Array(stringToByteArray('Hello'));
+  const wrongData = new Uint8Array(stringToByteArray('Hello0'));
+
+
+  it('should validate with the correct key and signature', () => {
+    cryptoVerifier.verify(data, signature, [rsaPubKey])
+      .then(isvalid => expect(isvalid).to.be.true);
+  });
+
+  it('should not validate with the correct key but wrong data', () => {
+    // Test with correct key, but wrong data.
+    cryptoVerifier.verify(wrongData, signature, [rsaPubKey])
```

Comment:
do these tests work correctly? we probably need do an explicit return here for the `it` block to actually wait for the asynchronous task.

```js
  return cryptoVerifier.verify(...)
      .then(...)
```